topologies/binding/binding.go

```diff
@@ -197,27 +197,63 @@ func (d *staticDUT) DialP4RT(ctx context.Context, opts ...grpc.DialOption) (p4pb
 
 func (d *staticDUT) DialCLI(context.Context) (binding.CLIClient, error) {
 	sshOpts := d.r.ssh(d.dev)
-	c := &ssh.ClientConfig{
+	config := &ssh.ClientConfig{
 		User: sshOpts.Username,
 		Auth: []ssh.AuthMethod{
 			ssh.Password(sshOpts.Password),
 			ssh.KeyboardInteractive(sshInteractive(sshOpts.Password)),
 		},
 	}
+	sc, err := createSSHClient(config, sshOpts)
+	if err != nil {
+		return nil, err
+	}
+	return newCLI(sc)
+}
+
+func (d *staticDUT) DialSSH(_ context.Context, sshAuth binding.SSHAuth) (binding.SSHClient, error) {
+	var config *ssh.ClientConfig
+	switch auth := sshAuth.(type) {
+	case binding.PasswordAuth:
+		config = &ssh.ClientConfig{
+			User: auth.User,
+			Auth: []ssh.AuthMethod{
+				ssh.Password(auth.Password),
+				ssh.KeyboardInteractive(sshInteractive(auth.Password)),
+			},
+		}
+	case binding.KeyAuth:
+		signer, err := ssh.ParsePrivateKey(auth.Key)
```

Comment:
This will probably not work with our use case since we may use a key chain.

For our use case, can you do: `signer, err := ssh.Unmarshal(auth.Key, ssh.Signer)`?